packages/web3-utils/src/index.js

```diff
@@ -328,36 +328,52 @@ var toChecksumAddress = function (address) {
  * @returns {Number} -1, 0, or 1
  */
 var compareBlockNumbers = function(a, b) {
+    // Increasing order: (genesis = earliest), safe, (finalized ~ latest), pending
+    // safe vs block-num cant be compared as block number provided can be on left or right side of safe tag, until safe tag block number is extracted and compared
```

Comment:
So, what would this method return if `a` or `b` was `safe` and the other one was a number? Throw an exception or return a string `unknown`? Or what do you think?

Reply:
We can do that. But it could be a little too aggressive. What about returning `undefined` as the function is not able to decide?

Reply:
now function will return undefined for safe vs non zero num